.github/workflows/goreleaser.yml

```diff
@@ -67,3 +69,36 @@ jobs:
     with:
       base64-subjects: "${{ needs.goreleaser.outputs.hashes }}"
       upload-assets: true # upload to a new release
+  verification:
+    needs: [goreleaser, provenance]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # To add assets to a release.
```

Comment:
please file a bug on https://github.com/actions/runner and cc me. Requiring write permission is really odd

Reply:
let me try it again, I changed back it to "read".

Reply:
see it didn't worked: _https://github.com/developer-guy/osv-scanner/actions/runs/4519224365/jobs/7959640924_

Reply:
it didn't work with "write" permission either: 
_https://github.com/developer-guy/osv-scanner/actions/runs/4519341906/jobs/7959797410#step:1:17_

but it worked once what is going on 🤔 
_https://github.com/developer-guy/osv-scanner/actions/runs/4504207210/jobs/7928457868#step:1:17_

Reply:
this script is just working fine with `write` permission: 

```bash
set -euo pipefail
  TAG=${GITHUB_REF#refs/tags/}
  echo "Downloading assets for tag $TAG"
  for i in $(gh api "repos/$GITHUB_REPOSITORY/releases" | jq -r '.[] | select(.tag_name=='\"$TAG\"').assets[].id')
  do
    echo "Downloading asset $i"
    curl -L -o $(gh api "repos/$GITHUB_REPOSITORY/releases/assets/$i" | jq -r '.name') -H "Authorization: token $GH_TOKEN" -H "Accept: application/octet-stream" $(gh api "repos/$GITHUB_REPOSITORY/releases/assets/$i" | jq -r '.url')
  done
  echo "Downloading provenance $PROVENANCE"
  gh -R $GITHUB_REPOSITORY release download --clobber $GITHUB_REF_NAME -p $PROVENANCE
```

Reply:
we even don't need this line:

```bash
gh -R $GITHUB_REPOSITORY release download --clobber $GITHUB_REF_NAME -p $PROVENANCE
```

and I also disabled uploading assets in the generic generator settings:

```bash
upload-assets: false
```